runtime-common/loader.ts

```diff
@@ -0,0 +1,212 @@
+// @ts-ignore
+import TransformModulesAmd from "@babel/plugin-transform-modules-amd";
+import { transformSync } from "@babel/core";
+import { Deferred } from "./deferred";
+import type { Realm } from "@cardstack/runtime-common/realm";
+import { RealmPaths } from "./paths";
+import { baseRealm } from "@cardstack/runtime-common";
+
+type RegisteredModule = {
+  state: "registered";
+  dependencyList: string[];
+  implementation: Function;
+};
+
+type FetchingModule = {
+  state: "fetching";
+  deferred: Deferred<unknown>;
+};
+
+type Module =
+  | RegisteredModule
+  | FetchingModule
```

Comment:
I added this new state to the module loader (adopted from cardhost's local realm) to represent the state during which we are downloading the module src